Add troubleshooting guidance for CSPs without unsafe-inline (#2040)
withastro/docs: src/pages/en/guides/troubleshooting.md

**Status**: Expected Astro behavior, as intended.

### Refused to execute inline script

You may see the following error logged in the browser console:

> Refused to execute inline script because it violates the following Content Security Policy directive: …

> This means that your site’s [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) (CSP) disallows inline scripts. Because Astro uses inline `<script>` tags in its output by default, interactive components may break triggering this error.

Reviewer: ```suggestion
This means that your site’s [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) (CSP) disallows inline scripts, which are Astro's default `<script>` output.
```
Maybe this is all that needs to be said?

Author: Maybe this? Because it could also be a framework component rather than a `<script>` tag that causes the `<script>` in the build output. Only hesitancy is that this clashes with “zero JS by default”, but I think it’s OK in this context.

```suggestion
This means that your site’s [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) (CSP) disallows running inline `<script>` tags, which Astro outputs by default.
```

Reviewer: Looks great! (to me)
